packages/shared_preferences/shared_preferences_foundation/lib/shared_preferences_foundation.dart

```diff
@@ -2,106 +2,236 @@
+  ) async {
+    return _convertKnownExceptions<double>(() async => await _api.getValue(
+        key, _convertOptionsToPigeonOptions(options)) as double?);
   }
 
   @override
-  Future<bool> remove(String key) async {
-    await _api.remove(key);
-    return true;
+  Future<int?> getInt(
+    String key,
+    SharedPreferencesOptions options,
+  ) async {
+    return _convertKnownExceptions<int>(() async => await _api.getValue(
+        key, _convertOptionsToPigeonOptions(options)) as int?);
   }
 
   @override
-  Future<bool> setValue(String valueType, String key, Object value) async {
-    final _Setter? setter = _setters[valueType];
-    if (setter == null) {
-      throw PlatformException(
-          code: 'InvalidOperation',
-          message: '"$valueType" is not a supported type.');
+  Future<List<String>?> getStringList(
+    String key,
+    SharedPreferencesOptions options,
+  ) async {
+    // TODO(tarrinneal): Remove cast once https://github.com/flutter/flutter/issues/97848
+    // is fixed. In practice, the values will never be null, and the native implementation assumes that.
+    return _convertKnownExceptions<List<String>>(() async =>
+        ((await _api.getValue(key, _convertOptionsToPigeonOptions(options)))
+                as List<Object?>?)
+            ?.cast<String>());
+  }
+
+  @override
+  Future<void> clear(
+    ClearPreferencesParameters parameters,
+    SharedPreferencesOptions options,
+  ) async {
+    final PreferencesFilters filter = parameters.filter;
+    return _convertKnownExceptions<void>(() async => _api.clear(
+          filter.allowList?.toList(),
+          _convertOptionsToPigeonOptions(options),
+        ));
+  }
+
+  @override
+  Future<Map<String, Object>> getPreferences(
+    GetPreferencesParameters parameters,
+    SharedPreferencesOptions options,
+  ) async {
+    final PreferencesFilters filter = parameters.filter;
+    final Map<String?, Object?>? data =
+        await _convertKnownExceptions<Map<String?, Object?>>(
+            () async => _api.getAll(
+                  filter.allowList?.toList(),
+                  _convertOptionsToPigeonOptions(options),
+                ));
+
+    return data!.cast<String, Object>();
+  }
+
+  Future<T?> _convertKnownExceptions<T>(Future<T?> Function() method) async {
+    try {
+      final T? value = await method();
+      return value;
+    } on PlatformException catch (e) {
+      if (e.code == _argumentErrorCode) {
+        throw ArgumentError(
+            'shared_preferences_foundation argument error ${e.message ?? ''}');
+      } else {
+        rethrow;
+      }
+    }
+  }
+}
+
+/// Options for the Foundation specific SharedPreferences plugin.
+@immutable
+class SharedPreferencesFoundationOptions extends SharedPreferencesOptions {
+  /// Creates a new instance with the given options.
+  SharedPreferencesFoundationOptions({
+    this.suiteName,
+  }) {
+    if (Platform.isIOS && !(suiteName?.startsWith('group.') ?? true)) {
```

Comment:
Let's comment why here, so it's obvious why we're restricting the API. Something like:

```dart
// Ensure that use of suite is compliant with required reason API category 1C8F.1; see
// https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api
```